### PR TITLE
Rewrite tracker_utils to use urlparse

### PR DESCRIPTION
**Pull request by vandenheuvel**
This fixes https://github.com/Tribler/tribler/issues/2817. 

I choose to use `urlparse` from the standard libraries. It will be easy to migrate this code to python3 as an equivalent standard library is available for that version.

### CODE REVIEW

**Comment by devos50** [comment]
ok to test

**Comment by vandenheuvel** [comment]
Does someone know why tests on Linux are failing?

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

File: Tribler/Core/Utilities/tracker_utils.py [hunk 1]
```diff
@@ -1,115 +1,101 @@
...
-#    udp://tracker.openbittorrent.com:80
-#    http://tracker.openbittorrent.com:80/announce
-# ------------------------------------------------------------
 def get_uniformed_tracker_url(tracker_url):
+    """
+    Parse a tracker url of basestring type.
+
+    The following checks and transformations are applied to the url:
+        - Check if the url is valid unicode data
+        - Strip whitespace
```
Comment (devos50) [inline Tribler/Core/Utilities/tracker_utils.py]: *whitespaces

File: Tribler/Core/Utilities/tracker_utils.py [hunk 2]
```diff
@@ -1,115 +1,101 @@
...
+        - Strip a trailing '/'
+        - Check that port is
```
Comment (devos50) [inline Tribler/Core/Utilities/tracker_utils.py]: 'that the port...'

File: Tribler/Test/Core/Utilities/test_tracker_utils.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
+from nose.tools import raises
+
+from Tribler.Core.Utilities.tracker_utils import parse_tracker_url, get_uniformed_tracker_url, \
+    MalformedTrackerURLException
+
+from Tribler.Test.test_as_server import BaseTestCase
+
+
+class TestGetUniformedTrackerUrl(BaseTestCase):
```
Comment (devos50) [inline Tribler/Test/Core/Utilities/test_tracker_utils.py]: Please make this a subclass of `TriblerCoreTest`.

File: Tribler/Test/Core/Utilities/test_tracker_utils.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        result = get_uniformed_tracker_url("http://torrent.ubuntu.com:6969/announce")
+        self.assertEqual(result, u'http://torrent.ubuntu.com:6969/announce')
+
+    def test_uniform_scheme_correct_http_training_slash(self):
+        result = get_uniformed_tracker_url("http://torrent.ubuntu.com:6969/announce/")
+        self.assertEqual(result, u'http://torrent.ubuntu.com:6969/announce')
+
+    def test_uniform_scheme_unknown(self):
+        result = get_uniformed_tracker_url("unknown://tracker.openbittorrent.com/announce")
+        self.assertEqual(result, None)
```
Comment (devos50) [inline Tribler/Test/Core/Utilities/test_tracker_utils.py]: Please use `assertIsNone`

File: Tribler/Test/Core/Utilities/test_tracker_utils.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    def test_uniform_http_default_port(self):
+        result = get_uniformed_tracker_url("http://torrent.ubuntu.com/announce")
+        self.assertEqual(result, u'http://torrent.ubuntu.com/announce')
+
+    def test_uniform_http_default_port_given(self):
+        result = get_uniformed_tracker_url("http://torrent.ubuntu.com:80/announce")
+        self.assertEqual(result, u'http://torrent.ubuntu.com/announce')
+
+
+class TestParseTrackerUrl(BaseTestCase):
```
Comment (devos50) [inline Tribler/Test/Core/Utilities/test_tracker_utils.py]: Again, please subclass from `TriblerCoreTest`.

**Comment by vandenheuvel** [comment]
@devos50 Thanks for the review.

**Comment by devos50** [comment]
@vandenheuvel almost ready, please clean up your commit history so you don't have fixes of fixes. I.e. you can have two commits where in one commit, you perform the refactoring of the tracker_utils file and in the other commit, you add the tests.

**Comment by vandenheuvel** [comment]
retest this please

**Comment by vandenheuvel** [comment]
@devos50 Are you sure I have permissions to retest?

**Comment by devos50** [comment]
It should work correctly?

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@vandenheuvel please rebase against `devel` instead of merging.

**Pull request merged**